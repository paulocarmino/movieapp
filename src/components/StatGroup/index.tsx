
export type StatGroupProps = {
  icon: JSX.Element
  title: string
  value: string
}

const StatGroup = ({ icon, title, value }: StatGroupProps) => (
  <span className="group inline-flex relative z-0 max-h-[38px] rounded-md shadow-sm">
    <div
      className="inline-flex relative focus:z-10 items-center py-2 px-4 text-xs font-medium text-gray-100 rounded-l-md border border-gray-100 focus:outline-none lg:text-sm"
    >
      <div className="mr-2 -ml-1 text-gray-100">
        {icon}
      </div>
      {title}
    </div>
    <div
      className="inline-flex relative focus:z-10 items-center py-2 px-3 -ml-px text-sm font-medium text-gray-100 rounded-r-md border border-gray-100 focus:outline-none"
    >
      {value}
    </div>
  </span>
)

export default StatGroup
