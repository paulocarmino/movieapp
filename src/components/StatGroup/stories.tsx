import { Story, Meta } from '@storybook/react'
import { MdOutlinePoll } from 'react-icons/md'

import StatGroup, { StatGroupProps } from '.'

export default {
  title: 'StatGroup',
  component: StatGroup,
  argTypes: {
    title: {
      type: 'string'
    },
    value: {
      type: 'string'
    }
  },
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta


export const Default: Story<StatGroupProps> = (args) => <StatGroup {...args} icon={<MdOutlinePoll size={20} />} />

Default.args = {
  title: "Vote Average",
  value: "7.9"
}
