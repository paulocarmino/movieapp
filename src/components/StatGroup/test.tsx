import { render, screen } from '@testing-library/react'
import { MdOutlinePoll } from 'react-icons/md'

import StatGroup from '.'

describe('<StatGroup />', () => {
  it('should render the heading', () => {
    const { container } = render(<StatGroup title="Popularity" value="82.3%" icon={<MdOutlinePoll size={20} />} />)

    expect(screen.getByText('Popularity')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
