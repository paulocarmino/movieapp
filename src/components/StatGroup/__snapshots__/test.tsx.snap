// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatGroup /> should render the heading 1`] = `
<span
  class="group inline-flex relative z-0 max-h-[38px] rounded-md shadow-sm"
>
  <div
    class="inline-flex relative focus:z-10 items-center py-2 px-4 text-xs font-medium text-gray-100 rounded-l-md border border-gray-100 focus:outline-none lg:text-sm"
  >
    <div
      class="mr-2 -ml-1 text-gray-100"
    >
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM7 10h2v7H7zm4-3h2v10h-2zm4 6h2v4h-2z"
        />
      </svg>
    </div>
    Popularity
  </div>
  <div
    class="inline-flex relative focus:z-10 items-center py-2 px-3 -ml-px text-sm font-medium text-gray-100 rounded-r-md border border-gray-100 focus:outline-none"
  >
    82.3%
  </div>
</span>
`;
